rust_src/src/frames.rs

```diff
@@ -1,14 +1,57 @@
 //! Generic frame functions.
 
+use std::ptr;
+
 use remacs_macros::lisp_fn;
-use remacs_sys::Lisp_Frame;
-use remacs_sys::selected_frame as current_frame;
+use remacs_sys::{selected_frame as current_frame, Lisp_Frame};
+use remacs_sys::Qframe_live_p;
+use remacs_sys::fget_terminal;
 
 use lisp::{ExternalPtr, LispObject};
 use lisp::defsubr;
 
 pub type LispFrameRef = ExternalPtr<Lisp_Frame>;
 
+impl LispFrameRef {
+    #[inline]
+    pub fn is_live(&self) -> bool {
+        unsafe { fget_terminal(self.as_ptr()) != ptr::null() }
```

Comment:
should be using `ptr.is_null()` method instead?

Reply:
Good question. I will have to try it and see what happens.

You are suggesting:

    ! fget_terminal(self.as_ptr()).is_null()

correct?

Reply:
yep